scripts/espresso_block_scanner.py

```diff
@@ -0,0 +1,111 @@
+### Simple live Espresso block scanner
+### Scan all blocks looking for transaction on each VM. 
+### For each found transaction, display the VM block containing that transaction, including inclusion proofs for transactions in the Espresso block.
+
+import urllib.request
+import json;
+import time;
+import datetime;
+import sys;
+host="localhost"
+port="50000"
+if len(sys.argv) > 1:
+	host=sys.argv[1]
+if len(sys.argv) > 2:
+	port=sys.argv[2]
+url="http://"+host+":"+port+"/availability/block/"
+
+# Keep track of transaction to avoid showing duplicates
+payloads = set()
+
+# Scan block by block
+block_number = 0
+while True:
+	try:
+		page = urllib.request.urlopen(url+str(block_number))
+	except urllib.error.HTTPError as e:
+		# "Block not produced yet. waiting for 1 second
+		time.sleep(1)
+		continue;
+	except urllib.error.URLError as ex:
+		# Server is down, reset block number to 0
+
+		print("URLERROR trying again in 10 secs")
+		print(ex)
+		time.sleep(10)
+		block_number = 0
+	else:
+		# We are Ok Espresso block found in page
+		# If block contains transactions, extract Rollup id and find block with inclusion proofs for the rollup
+
+		block = page.read()
+		struct = json.loads(block)
+
+		# throw exception if json is different from expected
+		try:
+			txns = struct['block']['transaction_nmt']
+			# we only need to show the block once per vm, so if several transactions are seen for same vm in same block, treat them as duplicates
+			seen_vms = set();
+			for txn in txns:
+				vm = txn['vm'] # extract Rollup ID
+				payload = payload_hex=bytearray(txn['payload']).hex()
```

Comment:
The `payload_hex` looks unused. 

If the bytes are not mutated `bytes` can be used in state of `bytearray`.